pybamm/models/full_battery_models/lithium_ion/basic_dfn_half_cell.py

```diff
@@ -0,0 +1,420 @@
+#
+# Basic Doyle-Fuller-Newman (DFN) Half Cell Model
+#
+import pybamm
+from .base_lithium_ion_model import BaseModel
+
+
+class BasicDFNHalfCell(BaseModel):
+    """Doyle-Fuller-Newman (DFN) model of a lithium-ion battery with lithium counter
+    electrode, from [2]_.
+
+    This class differs from the :class:`pybamm.lithium_ion.DFN` model class in that it
+    shows the whole model in a single class. This comes at the cost of flexibility in
+    comparing different physical effects, and in general the main DFN class should be
+    used instead.
```

Comment:
rephrase this to make it clear that it's half-cell

Reply:
Done